pkg/nsx/services/common/wrap.go

```diff
@@ -0,0 +1,53 @@
+package common
+
+import (
+	"github.com/openlyinc/pointy"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+// WrapInfra TODO(gran) refactor existing code in other package
+func (service *Service) WrapInfra(children []*data.StructValue) (*model.Infra, error) {
+	// This is the outermost layer of the hierarchy infra client.
+	// It doesn't need ID field.
+	resourceType := ResourceTypeInfra
+	infraObj := model.Infra{
+		Children:     children,
+		ResourceType: &resourceType,
+	}
+	return &infraObj, nil
+}
+
+func (service *Service) WrapVPC(vpc *model.Vpc) ([]*data.StructValue, error) {
```

Comment:
Why not put WrapVPC and WrapLBS in pkg/nsx/services/vpc/wrap.go?

Reply:
Although these 2 functions are used by VPC only, they can be used by other packages in future.
In future, I think we might refactor these functions to one function which can wrap multiple types to Children resource.